code/controllers/subsystem/achievements.dm

```diff
@@ -59,10 +59,9 @@ SUBSYSTEM_DEF(achievements)
 				most_unlocked_achievement = instance
 	qdel(query)
 
-	for(var/i in GLOB.clients)
-		var/client/C = i
-		if(!C.player_details.achievements.initialized)
-			C.player_details.achievements.InitializeData()
+	for(var/client/C in GLOB.clients)
+		if(!C.persistent_client.achievements.initialized)
+			C.persistent_client.achievements.InitializeData()
```

Comment:
Wouldn't we want to change this to `for persistent_client in GLOB.persistent_clients`?

Reply:
i dunno how achievements work so i left it alone